content/kb/troubleshooting-manager-stuck-in-crash-loop-due-to-webhook-is-not-accessible.md

```diff
@@ -0,0 +1,50 @@
+---
+title: "Troubleshooting: Longhorn Manager Stuck in CrashLoopBackOff Due to Webhook Not Accessible"
+authors:
+- "Jack Lin"
+draft: false
+date: 2025-01-17
+versions:
+- ">= v1.5.0"
+categories:
+- "longhorn manager"
+---
+
+## Applicable versions
+
+Longhorn >= v1.5.0.
+
+## Symptoms
+
+Starting from v1.5.0, the webhook services were merged into the Longhorn Manager. During startup, the manager first initializes the admission and conversion webhook services, ensuring they are accessible by curling the webhook service's URL before starting the manager service.
+
+In some cases, the Longhorn Manager pod may enter a CrashLoopBackOff state due to the webhook service being inaccessible. Its failure can lead to the manager pod being repeatedly restarted. Below, we outline the three most common root causes for this issue and provide solutions to resolve it.
+
+### Root Cause 1: Firewall Is Not Set Correctly
+
+The firewall configuration may be preventing communication between the pods on different nodes in your Kubernetes cluster. This can block the Longhorn Manager from accessing the webhook service, resulting in the CrashLoopBackOff state.
+
+Please ensure that the pods on all nodes are able to communicate with each other. This can be verified by checking the firewall rules and ensuring that inter-pod communication is not blocked.
+
+### Root Cause 2: DNS Doesn't Work Correctly
+
+If DNS resolution is not functioning as expected, the Longhorn Manager may be unable to reach the webhook service by its DNS name. This is particularly important when accessing services through their internal Kubernetes DNS names.
+
+Please ensure that the dns works by executing into a pod and test if DNS resolution works correctly by attempting to curl the webhook service:
+
+```bash
+kubectl exec -it <pod-name> -- /bin/bash
+curl https://longhorn-conversion-webhook.longhorn-system.svc:9501/v1/healthz
+```
+
+Or check if CoreDNS or kube-dns service is running properly.
```

Comment:
Can we link this coreDNS troubleshooting ticket https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/

Reply:
sure, thanks for the link!

Reply:
Added, thanks!